responses/matchers.py

```diff
@@ -85,3 +85,32 @@ def match(request):
         return valid, ""
 
     return match
+
+
+def header_matcher(headers, allow_additional=True):
+    """
+    Matcher to match 'headers' argument in request using the responses library.
+    :param headers: (dict), same as provided to request
+    :param allow_additional: (bool), ignore and allow additional headers that
+                             were not specified in the matcher (default True).
+    :return: (func) matcher
+    """
+
+    def match(request):
+        request_headers = request.headers or {}
+
+        if allow_additional:
+            # filter down to just the headers specified in the matcher
+            request_headers = {k: v for k, v in request_headers.items() if k in headers}
+
+        valid = sorted(headers.items()) == sorted(request_headers.items())
+
+        if not valid:
+            return False, "%s doesn't match %s" % (
```

Comment:
```suggestion
            return False, "headers of %s doesn't match %s" % (
```

This might help make it more clear that headers were the source of the match failure.

Reply:
Fair, I can make that change. In practice you see messages like `"[('Authorization', 'foo')] doesn't match [('Authorization', 'bar')]"` so it's usually pretty obvious, but I have no qualms making it more obvious :)

I'll update the PR this evening

Reply:
I agree that it *should* be pretty simple, but if header matching is combined with parameters matching it could be ambiguous and a few extra words could save a developer some time down the road.

Reply:
Also, sorry for saying "this evening" 26 days ago 😳 that's bad time-management even for me 🤦‍♂️